lib/driver.js

```diff
@@ -271,12 +271,26 @@ class AndroidDriver extends BaseDriver {
       if (this.opts.fastReset) {
         await helpers.resetApp(this.adb, this.opts.app, this.opts.appPackage, this.opts.fastReset);
       }
+      if (this.opts.autoGrantPermissions) {
+        try {
+          await this.adb.grantAllPermissions(this.opts.appPackage, this.opts.app);
```

Comment:
and if it _does_ make sense, I don't like the duplicated code. we should factor out into a function and reuse.